### DISCUSSION TITLE
Error while building software on Mac

### DISCUSSION THREAD

**Discussion by VijayN10** [JacobBumgarner/VesselVio#75] [Q&A] [answered]
Hello, 

I am trying to install software on macOS (M1 Sonoma Version 14.3.1) and following the instructions given here https://jacobbumgarner.github.io/VesselVio/Build.html

While installing  python 3.8.8, I am getting the following message related to zlib

```
E-LOSXG891WFV:~ q21422vn$ pyenv install 3.8.8
python-build: use openssl from homebrew
python-build: use readline from homebrew
Downloading Python-3.8.8.tar.xz...
-> https://www.python.org/ftp/python/3.8.8/Python-3.8.8.tar.xz
Installing Python-3.8.8...
patching file 'Misc/NEWS.d/next/Build/2021-10-11-16-27-38.bpo-45405.iSfdW5.rst'
patching file configure
patching file configure.ac
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.3.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/q4/4btz2xfd5y72jv6gp5lzx_680000gp/T/python-build.20240221130732.8087
Results logged to /var/folders/q4/4btz2xfd5y72jv6gp5lzx_680000gp/T/python-build.20240221130732.8087.log

Last 10 log lines:
checking size of _Bool... 1
checking size of off_t... 8
checking whether to enable large file support... no
checking size of time_t... 8
checking for pthread_t... yes
checking size of pthread_t... 8
checking size of pthread_key_t... 8
checking whether pthread_key_t is compatible with int... no
configure: error: Unexpected output of 'arch' on OSX
make: *** No targets specified and no makefile found.  Stop.
E-LOSXG891WFV:~ q21422vn$ 

```

Could anyone please guide me on this?

Thanks.

**Comment by JacobBumgarner** [comment]
Unfortunately this issue seems specific to your installation of homebrew, pyenv, or something related on your M1 and is not related to VesselVio itself. I'm not able to help with this one, sorry. I would consider doing some googling around or possibly asking for help over at the `pyenv` GitHub page ([link](https://github.com/pyenv/pyenv)).

**Reply by VijayN10** [reply]
Thanks for your reply. I have asked help at `pyenv`  and waiting for their reply. 
Hope to get it resolved soon.

**Reply by VijayN10** [reply]
Hello @JacobBumgarner , 

I have asked this query to `pyenv`. However, they said that 3.8.8 doesn't support Arm64. (https://github.com/orgs/pyenv/discussions/2907#discussioncomment-8571406). 

Could you please help me how to proceed? I can am able to install version 3.8.10 with pyenv, Would you please confirm if VesselVio runs with 3.8.10?

Thanks.

**Reply by JacobBumgarner** [reply]
Afaik, VesselVio should be fine to run with Python 3.8.10. Please feel free to test this locally and update me on the results.

**Reply by VijayN10** [reply]
Thanks, I have installed 3.8.10 and then after running requirement script - 
`pip install -r requirements.txt`

I am getting following error - 

```
ERROR: Ignored the following versions that require a different python version: 0.42.0 Requires-Python >=3.9; 0.42.0rc1 Requires-Python >=3.9; 0.59.0 Requires-Python >=3.9; 0.59.0rc1 Requires-Python >=3.9; 1.25.0 Requires-Python >=3.9; 1.25.0rc1 Requires-Python >=3.9; 1.25.1 Requires-Python >=3.9; 1.25.2 Requires-Python >=3.9; 1.26.0 Requires-Python <3.13,>=3.9; 1.26.0b1 Requires-Python <3.13,>=3.9; 1.26.0rc1 Requires-Python <3.13,>=3.9; 1.26.1 Requires-Python <3.13,>=3.9; 1.26.2 Requires-Python >=3.9; 1.26.3 Requires-Python >=3.9; 1.26.4 Requires-Python >=3.9; 2.1.0 Requires-Python >=3.9; 2.1.0rc0 Requires-Python >=3.9; 2.1.1 Requires-Python >=3.9; 2.1.2 Requires-Python >=3.9; 2.1.3 Requires-Python >=3.9; 2.1.4 Requires-Python >=3.9; 2.2.0 Requires-Python >=3.9; 2.2.0rc0 Requires-Python >=3.9; 2.2.1 Requires-Python >=3.9; 3.2 Requires-Python >=3.9; 3.2.1 Requires-Python >=3.9; 3.2rc0 Requires-Python >=3.9; 3.8.0 Requires-Python >=3.9; 3.8.0rc1 Requires-Python >=3.9; 3.8.1 Requires-Python >=3.9; 3.8.2 Requires-Python >=3.9; 3.8.3 Requires-Python >=3.9
ERROR: Could not find a version that satisfies the requirement PyQt5==5.13.2 (from versions: 5.14.0, 5.14.1, 5.14.2, 5.15.0, 5.15.1, 5.15.2, 5.15.3, 5.15.4, 5.15.5, 5.15.6, 5.15.7, 5.15.8, 5.15.9, 5.15.10)
ERROR: No matching distribution found for PyQt5==5.13.2
```

**Reply by JacobBumgarner** [reply]
Unfortunately, I don't have the time to walk through every dependency issue with you. What you'll need to do is identify package versions that are compatible with Python 3.8.10.

I'll bite for this one to help you learn how to resolve these issues on your own: 

For PyQt5, a quick google of "PyQt5 python 3.8.10" led to this result:
https://github.com/pypa/packaging-problems/issues/681

It seems like there may be an issue with installing PyQt5 on ARM64 architectures. One approach will be trying to resolve this by installing the newest version of PyQt5 to see if that helps. If that doesn't work, you can do some googling to figure out how to get PyQt5 working on Apple silicon.

Best,
Jacob

**Reply by VijayN10** [reply]
Thanks Jacob, I will try to resolve these issues and update you here.

**Reply by LucaGiaccone** [reply, marked as answer]
@VijayN10 I encountered the same issues and managed to fix both. Here's what I did:

1. I created a virtual environment using Anaconda with Python 3.12.4.
2. I attempted to install the required packages, which gave me some warnings.
3. When I tried to run the `VesselVio.py` script, I encountered errors about missing packages, such as 'pyvista'. I manually installed all the missing packages one by one.
4. After this, I was able to successfully launch the app with Python 3.12.4.

However, I am not entirely sure if all the app's features will function correctly with Python 3.12.4. At least, the app launches now.

I hope this helps! I will keep you updated here about the software's performance.

**Reply by JacobBumgarner** [reply]
@LucaGiaccone Thanks for running through this on your Mac. Please feel free to make a PR if you are able to flush out the other package deps. 

My primary OS is now Windows (unfortunately...), so I'm unable to progress with further macOS support.

**Reply by VijayN10** [reply]
> running through this on your Mac. Please feel free to make a PR if you are able to flush out the other package deps.

Thanks @LucaGiaccone for the update!
